lib/base-generator.js

```diff
@@ -1,58 +1,70 @@
-        var diskFiles = globby.sync(sourcePath, globOptions);
-        return diskFiles.map(sourceFilePath => {
-            var commonPath = utils.getCommonPath(sourcePath);
-            var toFile = path.relative(commonPath, sourceFilePath);
-            var destPath = path.join(destinationPath, toFile);
-            destPath = !customOptions.preProcessPath ? destPath : customOptions.preProcessPath(destPath);
-            return { sourceFilePath, destPath };
-        });
-    }
-
-    _runPipeline(versionKey, destinationPath, pipeline) {
-        var patches = require(this.sourceRoot() + '/patch.json');
-        var rootPaths = ['default', ...(patches[versionKey] || [])];
-        pipeline.forEach(f => rootPaths.forEach(rootPath => f.bind(this)(rootPath, destinationPath)));
-    }
-
-    _baseGlobOptions() {
-        return {
-            dot: true,
-            sync: true,
-            debug: false,
-        };
-    }
+class BaseGenerator extends Generator {
+
+  // Decorates default copyTpl function. Allows preprocessing through the customOptions.preProcessPath function
+  _copyTpl(sourcePath, destinationPath, ctx, globOptions, customOptions) {
+    this._mapFiles(sourcePath, destinationPath, globOptions, customOptions)
+      .forEach(({ sourceFilePath, destPath }) => this.fs.copyTpl(sourceFilePath, destPath, ctx, globOptions));
+  }
+
+  // Decorates default copy function. Allows preprocessing through the customOptions.preProcessPath function
+  _copy(sourcePath, destinationPath, ctx, globOptions, customOptions) {
+    this._mapFiles(sourcePath, destinationPath, globOptions, customOptions)
+      .forEach(({ sourceFilePath, destPath }) => this.fs.copy(sourceFilePath, destPath, globOptions, ctx));
+  }
+
+  _mapFiles(sourcePath, destinationPath, globOptions, customOptions) {
+    const diskFiles = globby.sync(sourcePath, globOptions);
+    return diskFiles.map(sourceFilePath => {
+      const commonPath = utils.getCommonPath(sourcePath);
+      const toFile = path.relative(commonPath, sourceFilePath);
+      let destPath = path.join(destinationPath, toFile);
+      destPath = !customOptions.preProcessPath ? destPath : customOptions.preProcessPath(destPath);
+      return { sourceFilePath, destPath };
+    });
+  }
+
+  _updateFileContent(filePath, pipeline) {
+    let contents = this.fs.read(filePath);
+
+    contents = pipeline.reduce((contents, f) => f(contents), contents);
+
+    this.fs.write(filePath, contents);
+  }
+
+  _runPipeline(versionKey, destinationPath, pipeline) {
+    const patches = require(this.sourceRoot() + '/patch.json');
+    const rootPaths = ['default', ...(patches[versionKey] || [])];
+    pipeline.forEach(f => rootPaths.forEach(rootPath => f.bind(this)(rootPath, destinationPath)));
+  }
+
+  _baseGlobOptions() {
+    return {
+      dot: true,
+      sync: true,
+      debug: false,
+    };
+  }
 }
+
+module.exports = extensions.withConflictExclusions(BaseGenerator);
```

Comment:
Please use it as decorator

Reply:
It is not supported by Node at the moment